RIOT-OS/RIOT
# dist/tools/zsh-completion: provide completion for zsh

### Contribution description

This adds command completion that, when manually installed, provides RIOT-aware command completion for `make` in zsh. The completion supports:

- most commonly used `make` goals including a help text
- most commonly used `make` variables (such as `BOARD=`, `LTO=`, `BUILD_IN_DOCKER=`, `TOOLCHAIN=`) - possible values for these variables, including a list of boards detected at runtime to complete `BOARD=`

### Testing procedure

If you are a `zsh` user, you could follow the installation instructions and check

- RIOT-aware completion of `make` should happen in a RIOT repo
- regular completion of `make` should happen elsewhere

### Issues/PRs references

None
